### ISSUE TITLE
[3.5.0 RC1…RC3] Cannot upload (manual and auto-upload) bigger files any more (appears to be solved with RC4)

### ISSUE THREAD

**Issue by Wikinaut** [nextcloud/android#3551]
### Actual behaviour
Manual upload (and auto-upload) fails for bigger files, small files ~100 KB work

### Environment data
Android version:
8.0.0

Device model: 
HUAWEI P8 LITE 2017

Nextcloud app version:
Android 3.5.0 RC1 and RC2

Nextcloud server version:
15.0.2

**Comment by tobiasKaminsky** [comment]
Can you provide us additional infos via logcat? Or what is happening? https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat

**Comment by Wikinaut** [comment]
@tobiasKaminsky sorry. no, I cannot do this in the next days.

**Comment by tobiasKaminsky** [comment]
Next week we will release 3.5.0, so if this is a problem, we should try to fix it.
You can also use the dev version, which you can install in parallel

**Comment by AndyScherzinger** [comment]
If it is a general problem then it definitely needs to be fixed since upload is a mayor feature of the app. I also run 15.0.2 and 3.5.0RC* (well, `master`) but auto upload of images etc. is working fine for me.

**Comment by Wikinaut** [comment]
I have deinstalled the RC version and reinstalled the regular 3.4.6: works.
Going beta-testing 3.5.0 RC3: again the same problem.

There's something wrong. Somewhere.

**Comment by Wikinaut** [comment]
@AndyScherzinger The issue is not specific to auto-upload, it's "Cannot upload bigger files any more".

**Comment by AndyScherzinger** [comment]
@wikinaut @tobiasKaminsky I think this needs investigation before the stable release. @wikinaut: just a guess... Does it work for files smaller than 1MB on cellular networks and smaller than 10MB on wifi? If so than it means chunked uploads don't work for you while this could be a server configuration issue. Don't know for sure since @tobiasKaminsky is the expert on that matter and should be back online by tomorrow.

**Comment by Wikinaut** [comment]
@AndyScherzinger this was also my first test - I switched mobile data off - the effect was the same: I could not upload an image (~5 MB), but a cropped image (smaller file).

For you I will double check now, which again requires the de-,. re-, de- re-installation hell.

**Comment by AndyScherzinger** [comment]
You can also try it with the dev build via f-droid. Can be installed in parallel and is very close to the 3.5.0 RCs

**Comment by Wikinaut** [comment]
@AndyScherzinger Perhaps an interesting aspect: Uploads also fail on a totally different device (HUAWEI P8 Lite running 3.5.0 RC2), which is connected to a different user but on the same Nextcloud server.

**Comment by AndyScherzinger** [comment]
My guess is that it's somehow related to the server's configuration somehow. I am unfortunately afk for a while so let's hope Tobias can check this on Monday. He might need a test user on your server, mail would be Tobias at nextcloud dot com

**Comment by Wikinaut** [comment]
@AndyScherzinger 
I see these warnings since a while, but as the 3.4.x client correctly worked and works, I didn't fix them yet:

>Die PHP-Speichergrenze liegt unterhalb des empfohlenen Wertes von 512MB.

>Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/caldav" aufzulösen. Weitere Informationen findest Du in der Dokumentation.

>Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/carddav" aufzulösen. Weitere Informationen findest Du in der Dokumentation.

>Du verwendest derzeit PHP 7.0.33-0+deb9u1. Upgrade deine PHP-Version, um die Geschwindigkeits- und Sicherheitsupdates zu nutzen, welche von der PHP-Gruppe bereitgestellt werden, sobald diese Deine Distribution unterstützt.

**Comment by Wikinaut** [comment]
@AndyScherzinger 
I see these warnings since a while, but as the 3.4.x client correctly worked and works, I didn't fix them yet:

>Die PHP-Speichergrenze liegt unterhalb des empfohlenen Wertes von 512MB.

>Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/caldav" aufzulösen. Weitere Informationen findest Du in der Dokumentation.

>Dein Web-Server ist nicht richtig eingerichtet um "/.well-known/carddav" aufzulösen. Weitere Informationen findest Du in der Dokumentation.

>Du verwendest derzeit PHP 7.0.33-0+deb9u1. Upgrade deine PHP-Version, um die Geschwindigkeits- und Sicherheitsupdates zu nutzen, welche von der PHP-Gruppe bereitgestellt werden, sobald diese Deine Distribution unterstützt.

**Comment by tobiasKaminsky** [comment]
Can you create us a test account, test if the problem occurs also there and if so send the credentials to tobias at nextcloud dot com with a reference to this issue?

**Comment by tobiasKaminsky** [comment]
This would be really helpful, but I fear that this is a single configuration problem, as no one else reported back any upload problems.

**Comment by Wikinaut** [comment]
@tobiasKaminsky Where can I find your public key?

**Comment by tobiasKaminsky** [comment]
I got test credentials and tested it with emulator and Android 9, with 5mb and 90mb files, and had no problem.
So this must be unfortunately a combination of your account and device.

**Comment by bjo81** [comment]
@tobiasKaminsky As this issue appears at my device and also at @Wikinaut's device with his instance, it does not sound like an individual case.

**Comment by Wikinaut** [comment]
@tobiasKaminsky wrote
>I got test credentials and tested it with emulator and Android 9, with 5mb and 90mb files, and had no problem.
>So this must be unfortunately a combination of your account and device.

This is not true. I did not send you any information yet.

**Comment by tobiasKaminsky** [comment]
Hm. I got an email with credentials and it refers to this issue, it was by Bjoern (do not want to expose last name).

**Comment by bjo81** [comment]
@tobiasKaminsky It was me. And as I pointed out, I have the same issue.

**Comment by tobiasKaminsky** [comment]
This seems to happen due to whitespaces in userId.
@bjo81 is this also true for your non-working account?

**Comment by bjo81** [comment]
@tobiasKaminsky No, no whitespaces in the userId here.

**Comment by Wikinaut** [comment]
If this is really the case - I cannot simply change the username, I think - it's a perfidious regression.

**Comment by bjo81** [comment]
![screenshot_20190212-121942_nextcloud_dev](https://user-images.githubusercontent.com/1674578/52632062-c843dd00-2ec0-11e9-93c9-5e591166080f.png)
Tested again with the Nextcloud dev version from F-Droid. Manual upload works (the app just renames the file to foo (b).bar), while the automatic upload still complains about a conflict.

**Comment by tobiasKaminsky** [comment]
> If this is really the case - I cannot simply change the username, I think - it's a perfidious regression.

No, there is no need for action on users side. This can and needs to be fixed on client side.
@Wikinaut thank you for your help and test credentials!
I'll do a RC4, for testing today.


> Tested again with the Nextcloud dev version from F-Droid. Manual upload works (the app just renames the file to foo (2).bar), while the automatic upload still complains about a conflict.

This issue is only about upload problems. 
For Auto upload and sync conflicts please open up a new issue.

**Comment by bjo81** [comment]
> This issue is only about upload problems.
> For Auto upload and sync conflicts please open up a new issue.

Ok. First I thought that it could be also size related in my case, bu I think it's more like #3579

**Comment by tobiasKaminsky** [comment]
https://github.com/nextcloud/android-library/pull/245

**Comment by Wikinaut** [comment]
@tobiasKaminsky RC4: upload problem is still present

**Comment by tobiasKaminsky** [comment]
@Wikinaut this is quite strange as it is working now for me, also on your server with user "Tobias Nextcloud".
Can you give it a try with exact this user and upload a file bigger than 2 mb?
What is your error message?

**Comment by bjo81** [comment]
@tobiasKaminsky The user "Tobias Nextcloud" is on my instance.

**Comment by tobiasKaminsky** [comment]
@bjo81 you sent me a mail with tobias.nc and address is c.s.o
@Wikinaut sent me test credentials with "Tobias Nextcloud" and w.o.o
I tested it on later account.

**Comment by bjo81** [comment]
@tobiasKaminsky Sorry for the confusion, I thought you speak of the set Real Name.

**Comment by Wikinaut** [comment]
@tobiasKaminsky on my device, 10 minutes ago, around 10:30 CET today, a new download of RC4 startet suddendly, and immediately thereafter pending upload got executed. So it appears, at the first look right now, that the problem is solved in RC4. I will correct the issue title and close the issue.

**Comment by tobiasKaminsky** [comment]
Great to hear, @Wikinaut.
Maybe it was yesterday a delayed update to RC3.
You can always verify the version in settings dialog within our app.

**Comment by Wikinaut** [comment]
@AndyScherzinger Sagte ich schon "danke"?

**Comment by AndyScherzinger** [comment]
@wikinaut wofür?

**Comment by Wikinaut** [comment]
@AndyScherzinger Für das - nach der Meldung hier - dann doch schnelle Fixen.

**Comment by AndyScherzinger** [comment]
@wikinaut Der Dank gebührt @tobiasKaminsky! Ich hab nur die Prio hochgehalten, gefixt hat es Tobias ❤️

**Comment by Wikinaut** [comment]
@AndyScherzinger @tobiasKaminsky pls. also add usernames with spaces like "Tobias Nextcloud" to your test suites so that such a regression may never happen again. By the way, what's with usernames comprising Umlauts or in general UTF-8 characters?

**Comment by tobiasKaminsky** [comment]
https://github.com/nextcloud/server/blob/7b09d1192482093939027f45a633d5d0a12b7a38/lib/private/User/Manager.php#L316-L319

--> Allowed are: "a-z", "A-Z", "0-9" and "_.@-'" and whitespace

So no 😀 or üöä. This will be prohibited on creation.

But :+1: for adding user ids with whitespaces.

**Comment by Wikinaut** [comment]
@AndyScherzinger and @tobiasKaminsky please can you (again) ping and escalate issue https://github.com/nextcloud/desktop/issues/1018 which urgently need an (apparently similar) fix for the same issue on the desktop clients; it is a killer.
